File: Implementation/programmers/Lv.2/프린터.py
from collections import deque

def solution(priorities, location):
    answer = 0
    docu = []
    index = 0
    num = 0
    for priority in priorities :
        
        docu.append((index, priority))
        index += 1
    printer = deque(docu)
    
    while printer :
        max_pri = max(printer, key = lambda x : x[1])[1]
        
        if printer[0][1] == max_pri :
            num += 1
            if printer[0][0]  == location :
                answer = num
                break
            printer.popleft()
        else :    
            printer.rotate(-1)
    
    
    return answer

File: Programmers/Javascript/Lv.2/프린터.js
function solution(priorities, location) {
  let answer = 0;
  const printer = Array.from({ length: priorities.length }, (_, idx) => [
    priorities[idx],
    idx,
  ]);
  priorities.sort((a, b) => a - b);

  while (printer.length) {
    if (printer[0][0] === priorities[priorities.length - 1]) {
      const print = printer.shift();
      priorities.pop();
      answer++;
      if (print[1] === location) break;
    } else printer.push(printer.shift());
  }
  return answer;
}
